### PR TITLE
feat(sdk): file uploads, full AI Gateway, and a corrected API surface

### PR DESCRIPTION
**Pull request by Zeryther**
## Summary

Makes `@gigadrive/sdk` correct against the production API and ships a **production-grade file-upload client**, then rounds out the surface (pagination, hostnames, request logs) and the **full AI Gateway** (runtime + streaming + governance).

The SDK is open source; everything here describes only the **public HTTP contract** — no infra/vendor/internal-route details.

## File uploads (headline)

`client.applications.storage.upload({ … })` — one call that computes the required SHA-256, infers content type from the key, creates the session, and uploads the bytes resumably with progress, `AbortSignal` cancellation, and retries.

- Accepts browser `File`/`Blob`, Node `Buffer`/`Uint8Array`/`ArrayBuffer`, a Node file `path`, or a Node readable `stream`.
- `uploadBatch()` (bounded concurrency, per-file error isolation), `waitForCompletion()`, and low-level `uploadSessions.uploadToUrl()` / `resumeFromUrl()`.

## Correctness fixes (breaking, pre-1.0)

These calls were failing against production:

- **AI Gateway** now uses the correct `/ai/v1` base path (was `/v1` → 404).
- **Storage objects** are addressed by **object ID**; `getAccessUrl()` takes a `bucketId` + `expiresInSeconds`; object listing supports `prefix`/`delimiter`/`cursor`/`limit` and returns `commonPrefixes`/`nextCursor`; added `getByKey()`.
- **Upload-session create** now sends the required SHA-256 checksum (was 400).
- **Bucket create** requires `environmentId` and accepts an optional `slug`.
- Removed non-public `provider*` fields from `StorageBucket`/`StorageObject`; corrected `DeploymentStatus`.

## Full AI Gateway

Streamed chat completions (`chatCompletionsStream`), Responses, audio (speech/transcriptions), video generation + models, richer model metadata, and org-scoped governance (`organizations.aiGateway` usage/budgets/policies).

## Broader surface

Pagination (`page`/`perPage`/`cursor` + `paginate()`), application & deployment hostnames, application request logs, `HttpClient.put()`, auth-code `scopes`, and complete public exports.

## Verification

- ✅ `typecheck`, `eslint`, `prettier --check`
- ✅ `vitest` — **95 tests** pass (added checksum, streaming, pagination, upload-session, high-level upload, AI-gateway runtime + governance)
- ✅ `build` — CJS / ESM / IIFE + DTS
- ✅ Confidentiality scan clean (public contract only)
- 🔎 An adversarial multi-agent review was run; all 8 confirmed findings are fixed (SSE edge cases, 401 stream-body retry guard, abort-listener hygiene, video endpoint path/models, vendor wording).

⚠️ Not yet run: live smoke test against `api.gigadrive.network` (needs real `GIGADRIVE_*` credentials).

A `minor` changeset is included (`0.2.0`).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Production-grade file upload APIs with automatic checksum calculation, content-type inference, progress reporting, cancellation, and resumable uploads
  * Batch uploads with concurrency and per-file error isolation
  * Expanded AI Gateway “new surface”: streamed chat completions, responses, audio/video generation, and improved model metadata
  * Application request log access
  * Organization-scoped AI Gateway governance (budgets and policies)
  * Cursor/page pagination helper; complete SDK documentation (upload guide)

* **Breaking Changes**
  * AI Gateway base paths corrected to `/ai/v1`
  * Storage API correctness fixes (object addressing, listing parameters, and new `getByKey()`); signed access URL behavior updated
  * Upload sessions now require SHA-256; bucket creation now requires `environmentId` (optional `slug`)

### CODE REVIEW

File: packages/sdk/src/http-client.ts [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+  /** Opaque cursor for cursor-based pagination (from a previous response's `nextCursor`). */
+  cursor?: string;
+}
+
+/** @internal */
+export interface RequestOptions {
+  query?: Record<string, QueryValue>;
+  body?: unknown;
+  headers?: Record<string, string>;
+}
```
[the remark this answers is not in the document]
Comment (Zeryther) [reply packages/sdk/src/http-client.ts:42]: Added optional per-request `AbortSignal` support to `RequestOptions` (threaded into both fetch calls incl. the 401 retry), so callers can cancel/time-bound any request. I intentionally did **not** add a global default timeout: it would also wrap the streaming (`requestStream`/SSE) and large resumable-upload paths, where a fixed timeout would kill legitimately long-running operations. Resolving as the cancellation mechanism is now in place.
